### PR TITLE
[RFC 0167] nixpkgs maintainer requirements

### PR DESCRIPTION
**Pull request by adamcstephens**
This PR adds an expectation and requirement for nixpkgs maintainers to include two forms of contact information. The first is an account on the nixpkgs hosting platform (GitHub) and the second is an alternative form such as email, matrix and adds discourse.

[Rendered](https://github.com/adamcstephens/rfcs/blob/0167/rfcs/0167-maintainer-account.md)
[final maintainer-list.nix](https://github.com/adamcstephens/nixpkgs/blob/rfc0167/maintainers/maintainer-list.nix)

### CODE REVIEW

**Comment by infinisil** [comment]
This RFC has not acquired enough shepherds. This typically shows lack of interest from the community. In order to progress [a full shepherd team is required](https://github.com/NixOS/rfcs/#shepherd-team). Consider trying to raise interest by [posting in Discourse](https://discourse.nixos.org/), [talking in Matrix](https://matrix.to/#/#community:nixos.org) or reaching out to people that you know.

If not enough shepherds can be found in the next month we will close this RFC until we can find enough interested participants. The PR can be reopened at any time if more shepherd nominations are made.

[See more info on the Nix RFC process here](https://github.com/NixOS/rfcs/blob/master/README.md)

**Comment by Janik-Haag** [comment]
I would like to nominate myself as shepherd. I don't have prior experience shepherding a RFC, but I do know how to engage with the community and get things moving.

**Comment by piegamesde** [comment]
I nominate @RaitoBezarius, which I know has already spent a lot of time thinking about this topic

**Comment by RaitoBezarius** [comment]
> I nominate @RaitoBezarius, which I know has already spent a lot of time thinking about this topic

I humbly accept.

**Comment by kevincox** [comment]
I see two shepherds.

- @RaitoBezarius 
- @Janik-Haag

Looking for 1 more.

**Comment by JulienMalka** [comment]
> I see two shepherds.
> 
> 
> 
> - @RaitoBezarius 
> 
> - @Janik-Haag
> 
> 
> 
> Looking for 1 more.

I am willing to be a shepherd

File: rfcs/0167-maintainer-account.md
Comment (infinisil) [inline rfcs/0167-maintainer-account.md:7]:
```suggestion
shepherd-team: @Janik-Haag, @RaitoBezarius, @JulienMalka 
shepherd-leader: @Janik-Haag
```

File: rfcs/0167-maintainer-account.md
Comment (Aleksanaa) [inline rfcs/0167-maintainer-account.md:63]:
```suggestion
We would typically look at it if we notice that the author hasn't reacted to all unresolved package-related notifications for more than 3 months.
```
This paragraph made me a little confused: 

If there are several maintainers for a package, do they need to all appear when being pinged?
What if someone just missed one or two pings but still responds to others?

File: rfcs/0167-maintainer-account.md
Comment (Aleksanaa) [inline rfcs/0167-maintainer-account.md:252]:
```suggestion
This could be added for signoff of modifications to `maintainer-list.nix` only if it does provide the intendend behavior of granting nixpkgs permission to use data.
```

File: rfcs/0167-maintainer-account.md
Comment (adamcstephens) [reply rfcs/0167-maintainer-account.md:63]: I'm open to clarifying this, but the wording you're responding to is already in place: https://github.com/NixOS/nixpkgs/tree/e0cf8780daedd2bc39f5696a9cb5b834fe976009/maintainers#how-to-lose-maintainer-status

I do not expect all maintainers to appear for every package PR or Issue, but I do expect them to be involved frequently. I read this to mean they've not engaged in their responsibility _at all_ during the three months. If they respond to any pings then that would reset any counters. (Even though there are technically no "counters" here and would only be looked at if deemed a problem)

I'd probably suggest at least rewording the duration. Some packages have no activity in 3 months, so perhaps something along the lines of

> for more than 3 months, or 2 package releases, whichever is longer.

Comment (Aleksanaa) [reply rfcs/0167-maintainer-account.md:63]: Okay, changing this would be a non-goal

File: rfcs/0167-maintainer-account.md
Comment (Aleksanaa) [inline rfcs/0167-maintainer-account.md:262]:
```suggestion
This could be added for signoff of modifications to `maintainer-list.nix` only if it does provide the intendend behavior of granting nixpkgs permission to use data.
```

Oops

File: rfcs/0167-maintainer-account.md
Comment (sternenseemann) [inline rfcs/0167-maintainer-account.md:63]: This is very difficult to measure, as it is. Also what happens in the case of multiple maintainers? Does everyone have to reply to every notification in order to game the system?

**Review by sternenseemann — COMMENTED** [review]
This seems to introduce a lot of unnecessary bureaucracy with no real upside because it doesn't change what maintainers' responsibilities, despite being reachable which is in and of itself not very valuable.

I also doubt that this is really required since it is perfectly possible to remove maintainers from the list at the moment if it bothers anyone. Removing maintainers that have been inactive for a while if it is okay with the other maintainers is not a problem at all — no one is going to oppose that in practice.

Establishing a formal policy may just lead to unnecessary busywork where people start creating mass PRs of maintainer removals that creates a lot of noise for no real benefit.

File: rfcs/0167-maintainer-account.md
Comment (nyabinary) [reply rfcs/0167-maintainer-account.md:63]: Moreover, what about vacations, some maintainers can simply get burnt out for a while, and some packages release incredibly quickly as well, so what about those packages?

Comment (adamcstephens) [reply rfcs/0167-maintainer-account.md:63]: While I modified the language here slightly, I'd like to note that some of this is already in the existing README. https://github.com/NixOS/nixpkgs/tree/f0701b576aeea14c09352345cabad4a0b6652c59/maintainers#how-to-lose-maintainer-status

I will change this RFC to be a diff of this document as this isn't the first time this confusion has arisen.

**Comment by adamcstephens** [comment]
> This seems to introduce a lot of unnecessary bureaucracy with no real upside because it doesn't change what maintainers' responsibilities, despite being reachable which is in and of itself not very valuable.

I'd argue that being reachable is value and should be a minimum requirement of being a maintainer. How will you respond to issues with packages you maintain if you can't be reached?

One of the core tenants of this RFC is to establish a requirement for maintainers to have an account on the nixpkgs hosting platform (aka GitHub). This does have value as spelled out in the RFC.

File: rfcs/0167-maintainer-account.md
Comment (sternenseemann) [reply rfcs/0167-maintainer-account.md:63]: > this is already in the existing README

… which kind of illustrates that an RFC doesn't seem to be needed.

**Comment by adamcstephens** [comment]
> To me it looks like this is, if anything, a matter of clean up and amending the missing github account names to the maintainer list – both shouldn't require an RFC as well.

I would tend to agree, but unfortunately such changes have in the past not been well received and some do consider this a significant enough change to warrant the RFC process. I’m happy to modify the scope of what is currently written here, but unless something has changed some of this cannot be implemented by PR alone.

Relates to: https://github.com/NixOS/nixpkgs/pull/273220 https://github.com/NixOS/nixpkgs/pull/272199 https://github.com/NixOS/nixpkgs/pull/273146

File: rfcs/0167-maintainer-account.md
Comment (eclairevoyant) [inline rfcs/0167-maintainer-account.md:166]:
```suggestion
    Maintainers SHOULD provide at least one of the alternative forms of contact. The simplest option would be to provide
```
I disagree that maintainers MUST provide this info. As long as the maintainers are accessible on github, this is a non-issue; these are fallback forms of contact anyway, and they can be onerous requirements that serve to increase the barrier to contribution. I'm fine with "SHOULD".

(definitions are as per [IETF RFC 2119](https://www.rfc-editor.org/rfc/rfc2119).)

File: rfcs/0167-maintainer-account.md
Comment (eclairevoyant) [inline rfcs/0167-maintainer-account.md:77]:
```suggestion
```

Don't think we need to repeat this in every document, as the COC explicitly mentions "contributors and maintainers".

**Review by eclairevoyant — CHANGES_REQUESTED** [review]

File: rfcs/0167-maintainer-account.md
Comment (RaitoBezarius) [reply rfcs/0167-maintainer-account.md:166]: I cannot understand your objection, there is literally a trivial email address that is linked to your GitHub account that is public and already usable.

If you are saying that people put null email addresses in their GitHub account, I am not so sure if we have to support the onerous requirement for them to have a working way of being contacted over email.

Comment (RaitoBezarius) [reply rfcs/0167-maintainer-account.md:166]: The `MUST` is trivially fulfilled here.

File: rfcs/0167-maintainer-account.md
Comment (RaitoBezarius) [reply rfcs/0167-maintainer-account.md:77]: What is the cost of repeating it?

File: rfcs/0167-maintainer-account.md
Comment (eclairevoyant) [reply rfcs/0167-maintainer-account.md:166]: > I cannot understand your objection

Anonymity or, quite simply, not wanting to be contacted elsewhere. If maintainers don't respond to the GH notification, remove them as per the process already outlined.

> there is literally a trivial email address that is linked to your GitHub account that is public and already usable.

Untrue, gh provides anonymised, no-reply "committer" emails.

File: rfcs/0167-maintainer-account.md
Comment (eclairevoyant) [reply rfcs/0167-maintainer-account.md:77]: It's redundant, do we need to link to every policy in every other policy? The longer something is, the less peopele read it.

Comment (RaitoBezarius) [reply rfcs/0167-maintainer-account.md:77]: I don't think it's a big deal as this is at the end, if they don't read it, so be it.
This reason for not reading things does not matter, what matters is that it becomes policy and can be enforced.

File: rfcs/0167-maintainer-account.md
Comment (RaitoBezarius) [reply rfcs/0167-maintainer-account.md:166]: @eclairevoyant Indeed, I missed that, but then, this is a social project, if you cannot afford to be present on any other platform beyond GitHub and want to be extremely anonymous, I think it will be very hard to be a maintainer, but that's my personal opinion.

Is anonymity a possibility we want to offer for the model of maintainers we want?

Comment (eclairevoyant) [reply rfcs/0167-maintainer-account.md:166]: > if you cannot afford to be present on any other platform beyond GitHub ... I think it will be very hard to be a maintainer

That contradicts a point elsewhere in the RFC regarding the downsides of using email:

> Users and maintainers must leave the nixpkgs hosting platform in order to contact a maintainer, possibly fracturing communications and losing public history

Package or module discussions should stay public; everyone should be able to benefit from the conversation. This is not an encouragement to be antisocial; on the contrary, it's further cementing the expectation that maintainers use and respond via gh if they want to stay as maintainers.

If the maintainer is unresponsive, it also reduces the burden of the "initiator" to remove the maintainer, as the initiator only needs to reach out on one platform (rather than follow up on an email as well).

File: rfcs/0167-maintainer-account.md
Comment (adamcstephens) [reply rfcs/0167-maintainer-account.md:77]: I think it’s worthwhile to repeat here. This is where we describe how a maintainer can be removed, and it seems better to explain all the reasons rather than expect people to look in multiple places.

Comment (ryneeverett) [reply rfcs/0167-maintainer-account.md:77]: > This reason for not reading things does not matter, what matters is that it becomes policy and can be enforced.

Sorry if I've missed something since [the COC was introduced](https://discourse.nixos.org/t/new-code-of-conduct-discussion/35179/47), but it's hard not to read this as an another attempt to sneak a COC in through a side door. Isn't the current claim still that the COC is strictly applicable to the moderation team or has that changed?

Comment (adamcstephens) [reply rfcs/0167-maintainer-account.md:77]: It is my understanding that permanent bans by the moderation team also apply to GitHub. In which case this is already being applied. If I'm wrong, I'd be happy to be corrected, but if such a ban does block them from GitHub then they will not qualify as a maintainer any longer.

Comment (eclairevoyant) [reply rfcs/0167-maintainer-account.md:77]: > it seems better to explain all the reasons rather than expect people to look in multiple places.

I agree; the very next sentence explains the reason, which is why I didn't suggest removing that.

> Isn't the current claim still that the COC is strictly applicable to the moderation team or has that changed?

As per how I read the linked discussion, the COC is documenting (mostly) how the moderation team currently moderates. It would therefore apply to all participants in the community.

Comment (ryneeverett) [reply rfcs/0167-maintainer-account.md:77]: I don't think the current language is strictly inaccurate but it seems very easy to misinterpret. The harm of redundancy might be more obvious were it more precise:

> Maintainers are not immune to actions taken by the moderation team, such as applications of [their Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md), and must be held to the same standard as non-maintainers.

Comment (Janik-Haag) [reply rfcs/0167-maintainer-account.md:77]: As far as I'm aware, if the moderation team perma-bans someone they ban them on matrix, discourse and github. I don't really get the discussion about leaving the sentence in or not, the CoC applies anyways and as raito said having one sentence more in the document doesn't really cost anything.

File: rfcs/0167-maintainer-account.md
Comment (Janik-Haag) [reply rfcs/0167-maintainer-account.md:166]: > That contradicts a point elsewhere in the RFC regarding the downsides of using email:

> > Users and maintainers must leave the nixpkgs hosting platform in order to contact a maintainer, possibly fracturing communications and losing public history

It doesn't contradict each other. Yes reviews should be done over github so things are transparent.
But nix is a community project so having other ways to communicate feels like a good requirement, for example before I nominated people for commit bit I talked with them in private which wouldn't really be possible via github. And a lot of other things like discussing rfc implementation progress, developing new tools, talking about ideas happens via matrix or sometimes email, so having that as a requirement sounds reasonable.

File: rfcs/0167-maintainer-account.md
Comment (7c6f434c) [reply rfcs/0167-maintainer-account.md:77]: Technically, ban on GitHub causes inability to react, but indeed having this separate case permits applying changes based on announced long-term bans without waiting a couple of months here or a couple of weeks there. And as the item in the list is useful, one more sentence in it doesn't seem to harm much, indeed.

Comment (ryneeverett) [reply rfcs/0167-maintainer-account.md:77]: If I understand correctly, the reason we want this sentence is that somebody can't say, "*You can't ban me because then I can't fulfill my duties as enumerated in RFC 167!*" Fair enough, but I think the sentence as written is both inadequate to that purpose and misleading in other directions.

It's inadequate because AFAIK the COC is not necessarily comprehensive of all moderation rationale's. Somebody could still say, "*You can't ban me because according to RFC 167 only the COC can justify moderation of maintainers and I found some way to be annoying which weasels around the COC!*" Granted, it's hard to imagine the current COC not justifying any moderator action but a future COC might not be so vague. My point is that the COC would seem to be irrelevant to the moderation team's authority to ban people.

It's misleading because it uses the term "NixOS Code of Conduct" which implies that the NixOS community has adopted a COC which it has not. And it uses the legal term "immunity" which implies that the COC is a binding document, when the COC isn't even binding on moderators.

Language which is more to the point would avoid these issues:

```suggestion
Maintainers are not exempt from moderation actions, even if said actions inhibit or prevent them from fulfilling their duties as maintainers.
```

Comment (RaitoBezarius) [reply rfcs/0167-maintainer-account.md:77]: I disagree, I expect maintainers to adhere to the NixOS CoC. If you want to make this a blocker point, join the shepherd team.

Comment (piegamesde) [reply rfcs/0167-maintainer-account.md:77]: @ryneeverett's proposals of wording sound reasonable to me. I think that framing this in the context of moderation instead of CoC is a good idea.

Comment (RaitoBezarius) [reply rfcs/0167-maintainer-account.md:77]: After thinking more about it, I concede this point, I think the suggesting change is good.

File: rfcs/0167-maintainer-account.md
Comment (RaitoBezarius) [reply rfcs/0167-maintainer-account.md:166]: @eclairevoyant The problem is that you are having what I feel like is a virtual or an abstract concern about people not being able to provide an email address, and we are debating over that virtual concern.

Conversely, as a committer, as a release manager, I had multiple situations where I was not able to reach out to folks because they didn't provide a secondary contact point and that pinging them on GitHub was not efficient.

Maintainership has not been in very good shape, so I'd ask you to provide a little data panel of how many maintainers we have without a secondary contact point and what are the packages they are maintaining, other than that, I don't feel compelled to act upon those comments. Feel free to join the shepherd team if you care about this.

File: rfcs/0167-maintainer-account.md
Comment (7c6f434c) [reply rfcs/0167-maintainer-account.md:77]: The CoC wording as written would give RFC grounds to two annoying demands generally rejected by the moderation team — justify moderation actions specifically via CoC (note that currently CoC is a piece of documentation downstream from moderation policies…), and justify that «same standard» is applied. The proposed wording does fit better the setup installed by the moderation team RFC.

Comment (eclairevoyant) [reply rfcs/0167-maintainer-account.md:77]: > I think that framing this in the context of moderation instead of CoC is a good idea.

Just want to note that there is a following sentence, that was not touched here, that explicitly still mentions COC:

> Violations of the Code of Conduct severe enough to warrant permanent removal will incur removal as maintainer as well.

My feedback didn't apply to this sentence, but it seems like other feedback on this thread _does_.

File: rfcs/0167-maintainer-account.md
Comment (eclairevoyant) [reply rfcs/0167-maintainer-account.md:166]: > you are having what I feel like is a virtual or an abstract concern

There it is. Well, what I see is that you are replying to every concern with dismissiveness, and are therefore _assuming_ this is an abstract concern.

> I'd ask you to provide a little data panel of how many maintainers we have without a secondary contact point

If >99% of people have an email, then it proves that making it a requirement (MUST) won't help get maintainership in "good shape", and making it a recommendation (SHOULD) would have the same effect. If <1% of people have an email, then SHOULD can still improve the situation, and MUST is overkill. Either way, this data has nothing to do with my point.

Going back to the original motivation mentioned in the RFC:

> For example, a maintainer may choose not to provide their user information for the primary nixpkgs hosting platform.
> This opting out for one or more individuals can put extra maintenance overhead on the remaining maintainers; it fragments information which may be important for the overall community; and may even hide information from others working on nixpkgs.

It's quite clear that having a GH account, and setting up notifications accordingly, is necessary to be a maintainer on nixpkgs.

What I don't see is how having an email is "more efficient" when it takes longer to write an email than a GH comment, and requires a separate lookup to get the actual email (vs @-ing the gh username, which should almost always match the attrname in `lib.maintainers`)?

This is not a "virtual concern", it's a boundary. Nixpkgs is not every maintainer's life, nor does it necessarily have to be. If you choose to set your boundaries by saying "contact me on GH or these dozen other platforms" and be bombarded with this, then so be it, but forcing every maintainer to do so is both unreasonable and does not serve the purpose of ensuring maintainers are reachable. The main place we expect reachability for the responsibilities of maintainership is GH, so GH is the only MUST here. I do not want anonymous users bombarding my personal email _and_ GH just because some package build failed; let the notifications be in the **one public place** where maintainers have to be anyway.

Comment (RaitoBezarius) [reply rfcs/0167-maintainer-account.md:166]: > There it is. Well, what I see is that you are replying to every concern with dismissiveness, and are therefore assuming this is an abstract concern.

Well, yes, I am a bit tired of having the same things happening in all of my RFCs and debating what I feel like is non-concern at the scale of things we are operating.

> If >99% of people have an email, then it proves that making it a requirement (MUST) won't help get maintainership in "good shape", and making it a recommendation (SHOULD) would have the same effect. If <1% of people have an email, then SHOULD can still improve the situation, and MUST is overkill. Either way, this data has nothing to do with my point.

It proves nothing? There was no RFC before. If anything, this was all triggered by people proving they want clear policy on the matter because not having policy was deemed insufficient.

> It's quite clear that having a GH account, and setting up notifications accordingly, is necessary to be a maintainer on nixpkgs.

Interesting, it doesn't seem like everyone share this perspective, as this RFC was spawned off the drama of someone threatening GDPR enforcement because a GH account should not be required to contribute.

May I recommend reading up on the context surrounding this PR rather than doing assumptions?

> What I don't see is how having an email is "more efficient" when it takes longer to write an email than a GH comment, and requires a separate lookup to get the actual email (vs @-ing the gh username, which should almost always match the attrname in lib.maintainers)?

Because sometimes GitHub does not work or because we need to send private information (e.g. CVE handling when we do have embargoes, etc.).
It also acts as a backup if we don't have GitHub for any reason.

The "taking longer to write an email" than "a GH comment" is astounding to me, how do you know? I write emails faster than GH comments, given the brokenness of the GitHub UI.

(which should almost always match the attrname means that I have to lookup lib.maintainers and not once or twice, as a release manager BTW, so that's moot).

> This is not a "virtual concern", it's a boundary. Nixpkgs is not every maintainer's life, nor does it necessarily have to be. If you choose to set your boundaries by saying "contact me on GH or these dozen other platforms" and be bombarded with this, then so be it, but forcing every maintainer to do so is both unreasonable and does not serve the purpose of ensuring maintainers are reachable. 

Yes but this boundary you are imposing is something you want to see, for it to happen, you need to participate in the RFC process and work with the shepherds, if all you propose is to order us around to execute on your concerns, I am sorry this is not how RFCs work.

Yes, Nixpkgs is not every maintainer's life and I do not intend to make it so by asking two point of contact to every person who decide to take care of a package, otherwise, please make it clear how asking for email makes Nixpkgs the maintainer's life. Many other distroes have the **email** of their maintainer, I fail to understand how we are very different to them?

If your concern is about being bombarded by email as soon as you put the email, please consider creating an email address for contribution purpose, a lot of us do that, and it's not very hard, sometimes much smaller than maintaining packages. To be clear, spamming is a valid concern and if it came to our knowledge that someone is abusing those email addresses, that'd be a big problem.
In the meantime, I am maintainer of a bunch of things and my email is on there, I received 2 emails in one year regarding package maintenance, other than that, nothing. Maybe my experience is not representative though, maybe I don't maintain enough packages or high-profile packages, but here we go.

> The main place we expect reachability for the responsibilities of maintainership is GH, so GH is the only MUST here. I do not want anonymous users bombarding my personal email and GH just because some package build failed; let the notifications be in the one public place where maintainers have to be anyway.

Sure, I don't want to argue further with you in comments. I propose you two choices:

- Join the shepherd team and work with us regarding your concerns and my concerns and let's find an acceptable solution
- Refuse, but in that case, I am afraid I cannot action your comments

Comment (eclairevoyant) [reply rfcs/0167-maintainer-account.md:166]: > May I recommend reading up on the context surrounding this PR rather than doing assumptions?

I did read that thread, in its entirety, before responding here. I should have said "it's quite clear _to me_" as I assumed that was implied from context; nothing has universal agreement.

I'm not "ordering you around", that's a pretty bad-faith response to my comments. Anyway, if you want to be dismissive, so be it, I'll save my time. I'm not going to spend my free time setting up some Matrix server just to repeat the same things I said here, when I already spend enough time on nixpkgs.

Comment (RaitoBezarius) [reply rfcs/0167-maintainer-account.md:166]: > I'm not "ordering you around", that's a pretty bad-faith response to my comments. Anyway, if you want to be dismissive, so be it, I'll save my time. I'm not going to spend my free time setting up some Matrix server just to repeat the same things I said here, when I already spend enough time on nixpkgs.

Get me right, I don't want to be dismissive of your comments, but I spend also enough of my personal time in the Nix ecosystem (almost full-time?), the RFC process is that if this is a blocker for you, you can join the shepherd team, and we can have a call about this and try to work out the points.

If we cannot have this, I am sorry, but I had this in many of my RFCs and it's indistinguishable of me having to jump and have someone tell me "I want this, and I don't want to join the team to work with you", this is my definition of being ordered around. If this was not your intent, I conjure you to join the shepherd team, it doesn't require a Matrix server, AFAIK.

Comment (adamcstephens) [reply rfcs/0167-maintainer-account.md:166]: >  It also acts as a backup if we don't have GitHub for any reason.

In my opinion, this is the primary reason to have a second form of contact. I do not find it wise to solely rely on GitHub. It exists today and works for us, but if it doesn't tomorrow then I want the Nix/NixOS community to have a way to gather back together elsewhere.

> What I don't see is how having an email is "more efficient" when it takes longer to write an email than a GH comment, and requires a separate lookup to get the actual email (vs @-ing the gh username, which should almost always match the attrname in lib.maintainers)?

This sentence you're discussing actually doesn't even require email. It requires _one_ of the alternatives. Though the more this has been discussed, the more convinced I am that it should just be email and discourse/matrix should be dropped as fulfilling the second contact method.

> Is anonymity a possibility we want to offer for the model of maintainers we want?

I don't think so.

File: rfcs/0167-maintainer-account.md
Comment (adamcstephens) [reply rfcs/0167-maintainer-account.md:77]:
```suggestion
#### Community Moderation

Actions taken through community moderation can impact your ability to remain a maintainer. Maintainers who have been permanently banned from NixOS spaces through moderation will be also removed as maintainers.
```

File: rfcs/0167-maintainer-account.md
Comment (marsam) [inline rfcs/0167-maintainer-account.md:257]: Another unresolved question is what resources does NixOS offer to new and prospective maintainers?
For instance, Debian/Fedora/OpenSUSE have a mentoring program, unfortunately NixOS doesn't have one.

File: rfcs/0167-maintainer-account.md
Comment (adamcstephens) [reply rfcs/0167-maintainer-account.md:166]: I've opted to not make email the only alternative. If others want to advocate, I'm happy to change my mind.

File: rfcs/0167-maintainer-account.md
Comment (adamcstephens) [reply rfcs/0167-maintainer-account.md:77]: Done.

**Comment by adamcstephens** [comment]
For those following along, the changes in `maintainer-list.nix` are now presented as a diff. The final rendered version of this file is now linked in the original description and is here https://github.com/adamcstephens/nixpkgs/blob/rfc0167/maintainers/maintainer-list.nix

As working on this, I wondered if maybe we should also move all of this content to the README.md in `maintainers/` and put a reference note in the current location. This would allow for proper markdown rendering and niceties. :) Thoughts?

File: rfcs/0167-maintainer-account.md
Comment (kevincox) [reply rfcs/0167-maintainer-account.md:7]: @adamcstephens Can you please merge this change?

**Comment by kevincox** [comment]
RFCSC: @Janik-Haag is there anything blocking this PR moving forward? Can you post a status update, indicate the next steps or organize a shepherd meeting that can help move this RFC to completion.

**Comment by adamcstephens** [comment]
I don't plan to pursue this any further, and the shepherd has left the community. Feel free to adopt as desired.

**Pull request closed without merging**
